### PR TITLE
Update PopupService Implementation

### PR DESCRIPTION
**Pull request by TheCodeTraveler**
This PR uses the correct method: `IPopupService.ShowPopup<MyViewModel>()`

### CODE REVIEW

File: TestMauidotnet8/MainPage.xaml.cs [hunk 1]
```diff
@@ -24,12 +24,7 @@ private void OnCounterClicked(object sender, EventArgs e)
 
             SemanticScreenReader.Announce(CounterBtn.Text);
 
-            MyViewModel vm = new()
-            {
-                Text = "My modified text"
-            };
-
-            popupService.ShowPopup(vm);
+            popupService.ShowPopup<MyViewModel>();
```
Comment (TheCodeTraveler) [inline TestMauidotnet8/MainPage.xaml.cs:27]: In your `ContentPage`, where you do not have access to an instance of `MyViewModel`, you don't need to (and shouldn't) instantiate `MyViewModel` to pass it in as a parameter.

File: TestMauidotnet8/MyPopup.xaml.cs [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
     {
         InitializeComponent();
-
+        this.popupService = popupService;
         BindingContext = myViewModel;
     }
+    
+    void HandleButtonClicked(object? sender, EventArgs e)
+    {
+        popupService.ShowPopup((MyViewModel)BindingContext);
```
Comment (TheCodeTraveler) [inline TestMauidotnet8/MyPopup.xaml.cs:19]: When you _do_ have access to the Popup ViewModel that was instantiated by Dependency Injection, you can reference that instance and pass it into `ShowPopup(TViewModel);`

Comment (GuidoNeele) [reply TestMauidotnet8/MyPopup.xaml.cs:19]: Nobody is going to use it like that.

Comment (TheCodeTraveler) [reply TestMauidotnet8/MyPopup.xaml.cs:19]: This feedback is not helpful. 

Please help me understand why.

Comment (pajesz) [reply TestMauidotnet8/MyPopup.xaml.cs:19]: I use the onPresenting action to manipulate/populate the viewmodel with its data from service as needed:
`var res = await popupService.ShowPopupAsync<TViewModel>(
    (vm) => { 
        vm.LoadById(id); 
});`

**Pull request closed without merging**